src/main/java/com/khomishchak/goalsservice/service/scheduled/GoalsScheduledService.java

```diff
@@ -0,0 +1,52 @@
+package com.khomishchak.goalsservice.service.scheduled;
+
+import com.khomishchak.goalsservice.model.GoalType;
+import com.khomishchak.goalsservice.model.SelfGoal;
+import com.khomishchak.goalsservice.repository.SelfGoalRepository;
+
+import com.khomishchak.goalsservice.service.GoalsService;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class GoalsScheduledService {
+
+    private final GoalsService goalsService;
+
+    public GoalsScheduledService(GoalsService goalsService) {
+        this.goalsService = goalsService;
+    }
+
+    @Scheduled(cron = "0 0 0 * * ?")
+    public void doAtTheStartOfTheDay() {
+        List<SelfGoal> overdueGoals = goalsService.getAllOverdueGoals();
+        closeOverdueAndCreateNewGoals(overdueGoals);
+    }
+
+    private void closeOverdueAndCreateNewGoals(List<SelfGoal> overdueGoals) {
+        overdueGoals.forEach(overdueGoal -> {
+            closeOverdueGoal(overdueGoal);
+            SelfGoal newGoal = createNewFromOverdueGoal(overdueGoal);
+            goalsService.saveAll(List.of(overdueGoal, newGoal));
```

Comment:
Could you please:
1) fix n+1 problem
2) change the algorithm if it's possible: first of all collect all `overfueGoals` and then save all by single `saveAll()` not by invoking `saveAll(...)` for every `overdueGoal`